packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -196,6 +208,24 @@ class ThreadChannel extends Channel {
     return this.archiveTimestamp && new Date(this.archiveTimestamp);
   }
 
+  /**
+   * The time the thread was created at
+   * @type {?Date}
+   * @readonly
+   */
+  get createdAt() {
+    if (this.createdTimestamp !== null) {
+      return new Date(this.createdTimestamp);
+    }
+
+    // Private threads can rely on their ID for creation date
+    if (this.type === ChannelType.PrivateThread) {
+      return super.createdAt;
+    }
```

Comment:
You're not doing this for `createdTimestamp` so it will still be nullable regardless of the thread type

You could just assign `this.createdTimestamp` to `super.createdTimestamp` in the constructor in case the thread is private